interfaces/builtin/upower_control.go

```diff
@@ -0,0 +1,215 @@
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixProcessID
+   peer=(label=unconfined),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixUser
+   peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.UPower",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UPower{,/**}
+    interface=org.freedesktop.UPower*,
+
+# Allow traffic to/from org.freedesktop.DBus for the UPower service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UPower{,/**}
+    interface=org.freedesktop.DBus.*,
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/login1{,/**}
+    interface=org.freedesktop.DBus.*
+    peer=(label=unconfined),
+
+# Allow access to logind service as we need to query it for possible
+# power states and trigger these when the battery gets low and the
+# system enters a critical state.
+dbus (send)
+    bus=system
+    path=/org/freedesktop/login1{,/**}
+    interface=org.freedesktop.login1.Manager
+    member={CanPowerOff,CanSuspend,CanHibernate,CanHybridSleep,PowerOff,Suspend,Hibernate,HybrisSleep},
+    peer=(label=unconfined),
+`
+
+const upowerControlPermanentSlotSeccomp = `
+bind
+getsockname
+recvmsg
+sendmsg
+sendto
+recvfrom
+`
+
+const upowerControlPermanentSlotDBus = `
+<!-- DBus policy for upower (upstream version 0.99.4) -->
+<policy user="root">
+  <allow own="org.freedesktop.UPower"/>
+</policy>
+<policy context="default">
+
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.DBus.Introspectable"/>
+
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.DBus.Peer"/>
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.DBus.Properties"/>
+  <allow send_destination="org.freedesktop.UPower.Device"
+         send_interface="org.freedesktop.DBus.Properties"/>
+  <allow send_destination="org.freedesktop.UPower.KbdBacklight"
+         send_interface="org.freedesktop.DBus.Properties"/>
+  <allow send_destination="org.freedesktop.UPower.Wakeups"
+         send_interface="org.freedesktop.DBus.Properties"/>
+
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.UPower"/>
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.UPower.Device"/>
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.UPower.KbdBacklight"/>
+  <allow send_destination="org.freedesktop.UPower"
+         send_interface="org.freedesktop.UPower.Wakeups"/>
+</policy>
```

Comment:
Are all of these considered safe for everyone or just root? If everyone, then perhaps that is an argument for merging upower-control into upower-observe....